pkg/bff/auth/metadata.go

```diff
@@ -1,19 +1,47 @@
 package auth
 
-import "encoding/json"
+import (
+	"encoding/json"
+
+	"github.com/trisacrypto/directory/pkg/bff/models/v1"
+)
 
 // AppMetadata makes it easier to serialize and deserialize JSON from the auth0
 // app_metadata assigned to the user by the BFF (and ensures the data is structured).
 type AppMetadata struct {
-	OrgID string `json:"orgid"`
-	VASPs VASPs  `json:"vasps"`
+	OrgID         string              `json:"orgid"`
+	VASPs         VASPs               `json:"vasps"`
+	Organizations map[string]struct{} `json:"organizations"`
 }
 
 type VASPs struct {
 	MainNet string `json:"mainnet"`
 	TestNet string `json:"testnet"`
 }
 
+// TODO: Hash-based method might be more maintainable, but this avoids error handling for now
+func (meta *AppMetadata) Equals(other *AppMetadata) bool {
+	if meta.OrgID != other.OrgID {
+		return false
+	}
+
+	if meta.VASPs != other.VASPs {
+		return false
+	}
+
+	if len(meta.Organizations) != len(other.Organizations) {
+		return false
+	}
+
+	for i, org := range meta.Organizations {
+		if value, ok := other.Organizations[i]; !ok || value != org {
+			return false
+		}
+	}
```

Comment:
Not to be a buzzkill, but I make this mistake a lot too with map/set equality. This only checks to make sure that all the key/value pairs in `meta.Organizations` are in and equal to the key/value pairs in `other.Organizations`. If there are additional key/value pairs in `other.Organizations` then this method will return true. 

e.g. the following organizations will be equal with this logic

```
meta.Organizations := map[string]struct{}{
    "apple": {},
    "banana": {},
    "orange": {},
}

other.Organizations := map[string]struct{}{
    "apple": {},
    "banana": {},
    "orange": {},
    "cantaloupe": {},
    "frostbite": {},
}
```

Reply:
I had copilot generate this and I definitely thought it was wrong too, however the length check before the check you mentioned should cover that case. I'm going to change the organization map to a slice per your suggestion anyway though.

Reply:
Ah you're right about the length check - sorry about that!